### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by atnartur**
English and Russian documentaion for all methods of client.js and part methods of other js files.
Added npm task `docs` for generating docs via jsdoc.
Added instruction for docs generation to README.md.
Fixed some documentaion issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /node_modules
 /config.js
 /test
+/docs
```
Comment (qfox) [inline .gitignore]: eol at eof

File: README.md [hunk 2]
```diff
@@ -13,6 +13,14 @@ At first you need to install library
 npm install megaplanjs --save
 ```
 
+You can generate documentaion with commands:
+
+```
+npm install
+npm install jsdoc -g
```
Comment (qfox) [inline README.md]: Do we really need to install it globally? Can we use devDependencies or `"docs-deps": "npm i jsdoc'` subtask for that?

File: lib/megaplan/client.js [hunk 3]
```diff
@@ -35,6 +40,7 @@ var Client = function MegaplanClient (server, access_id, secret_key, success_cal
 	this.callbacks(o.success_callback, o.fail_callback);
 	this.auth2(o);
 };
+
 var clientProto = {
```
Comment (qfox) [inline lib/megaplan/client.js:44]: Did you tried to use `@lends` here?
http://usejsdoc.org/tags-lends.html

File: lib/megaplan/client.js [hunk 4]
```diff
@@ -50,12 +56,14 @@ var clientProto = {
...
 
 	/**
-	 * Sends a request
-	 * @class MegaplanClient
-	 * @method __request
+	 * Sends a request.
+	 * 
+	 * Посылает запрос.
+	 * 
+	 * @memberof MegaplanClient
```
Comment (qfox) [inline lib/megaplan/client.js]: ... instead of these `@memberof`s

File: lib/megaplan/client.js [hunk 5]
```diff
@@ -106,21 +118,32 @@ var clientProto = {
 	},
 
 	/**
-	 * @class MegaplanClient
+	 * @memberof MegaplanClient
 	 * @event auth
-	 * @params {Object} o hash with access_id, secrey_key, and some other auth information on success or error on fail 
+	 * @description
+	 * Params: Object - o hash with access_id, secrey_key, and some other auth information on success or error on fail.
```
Comment (qfox) [inline lib/megaplan/client.js:123]: Can you try?

```
@param {Object} o
@param {string} o.access_id
// etc...
```

File: lib/megaplan/client.js [hunk 6]
```diff
@@ -301,29 +346,32 @@ var clientProto = {
...
+	 * @todo Запрос на изменение дедлайна
+	 * @todo Принять или отклонить запрос на изменение дедлайна
+	 * @todo Изменение соисполнителей
+	 * @todo Изменение аудиторов
+	 * @todo Делегирование задачи
+	 * @todo Проверка прав на делегирование задачи
+	 * @todo Список пользователей, которым можно делегировать задачу
+	 * @todo Список задач и проектов, которые можно указывать как надзадачи для других задач
+	 * @todo Список проектов, которые можно указывать как надпроекты для других задач
+	 * @todo Конвертация задачи в проект
```
Comment (qfox) [inline lib/megaplan/client.js:355]: If we really want to do these — better to make issues on github ;-)

File: lib/megaplan/client.js [hunk 7]
```diff
@@ -106,21 +118,32 @@ var clientProto = {
...
+	 * Производит авторизацию клиента по паре логин-пароль.
+	 * Если авторизация успешна, то будет вызвано событие auth с объектом {access_id, secret_key}.
+	 * Эти значения могут использоваться позже для создания экземляра класса {MegaplanClient} самостоятельно.
+	 * 
+	 * [More info]{@link https://help.megaplan.ru/API_authorization}
+	 *
+	 * @memberof MegaplanClient
 	 * @param {String} login
-	 * @param {String} password
+	 * @param {String} password
```
Comment (qfox) [inline lib/megaplan/client.js]: Redundant trailing space

File: lib/megaplan/client.js [hunk 8]
```diff
@@ -245,30 +289,31 @@ var clientProto = {
...
-	 * @param {Object} [attaches] There is no support of attaches at the moment. Sorry.
-	 *   Должен содержать ключ add в котором находится массив с файлами
-	 *   @param {Array} add Массив приложенных файлов, должен передаваться POST-запросом
-	 *     @param {String} context Данные(контент файла), закодированные с использованием MIME base64
-	 *     @param {String} name Имя файла (будет фигурировать при выводе задачи)
+	 * @param {Object} [attaches] There is no support of attaches at the moment. Sorry. Должен содержать ключ add в котором находится массив с файлами
+	 * @param {Array} attaches.add Массив приложенных файлов, должен передаваться POST-запросом
+	 * @param {String} attaches.add.context Данные(контент файла), закодированные с использованием MIME base64
+	 * @param {String} attaches.add.name Имя файла (будет фигурировать при выводе задачи)
 	 * @return {MegaplanRequest}
```
Comment (qfox) [inline lib/megaplan/client.js]: Can you also replace `@return` with `@returns` everywhere?

File: lib/megaplan/client.js [hunk 9]
```diff
@@ -231,9 +271,13 @@ var clientProto = {
 
 	/**
 	 * Feedback (?)
-	 * @class
-	 * @method history
-	 * @param {String} message
+	 *
+	 * Обратная связь. (?)
```
Comment (qfox) [inline lib/megaplan/client.js]: Guess we can drop this `(?)`

File: lib/megaplan/client.js (hunk 6 above)
Comment (atnartur) [inline lib/megaplan/client.js:355]: Ok)
I want to do part of this tasks soon.

File: .gitignore (hunk 1 above)
Comment (qfox) [inline .gitignore]: Add here `\n`?

Comment (atnartur) [inline .gitignore]: Added

**Comment by qfox** [comment]
:+1:

**Pull request merged**